process-engine/bpmn-studio
# ✨ Add Open Diagrams Functionality

## Changes

1. Add Open Diagrams Functionality
  1.1 Save diagrams in local storage (diagrams don't have to be saved anymore when changing the diagram)
  1.2 Remove "Quit"- and "Leave"-modal
  1.3 Save viewbox of open diagrams
  1.4. Save selected element
2. Move Remote Solutions to the Bottom of the SolutionExplorer
3. Remove "Single Diagrams"

## Issues

Closes #374 

PR: #1516

## How to test the changes

- Open Diagram from Filesystem
- **Notice that they will be opened as "Open Diagrams"**
- Make some changes
- Change the diagram
- **Notice that the diagram will be marked with a circle in the solution explorer**
<img width="247" src="https://user-images.githubusercontent.com/20394992/59766297-2aae4f00-92a0-11e9-829b-f21f4c4b4d3e.png">

- Close the unsaved diagram
- **Notice that a modal will appear**
- Click Save
- **Notice that the diagram will be saved.**
- Click Don't Save
- **Notice that the diagram wont be saved**
- Click Cancel

- Open a diagram
- Make some changes
- Change the diagram
- Close the diagram with the changes
- **Notice that the diagram with the saves will be displayed with a modal asking if one wants to save**
- Confirm the Modal
- **Notice that the previous diagram will be displayed again**
